[Blog] Add the Pulsar 2022 Year in Review blog (#371)
apache/pulsar-site: blog/2023-01-10-pulsar-2022-year-in-review.md

---

> author: Apache Pulsar Community

Reviewer: ```suggestion
author: Sherlock Xu
```
You deserve it 😊

Author: Thank you, but I refuse to do this. This is a summary blog for Pulsar achievements over the past year made possible by the community.

- [Pulsar Summit Asia 2022](pathname:///blog/2022/12/01/pulsar-summit-asia-2022-recap/): With 1,500+ registrations and 40,000+ views globally, this two-day virtual event gathered 41 speakers giving 3 keynote addresses and 36 sessions. They came from companies like ByteDance, Huawei, Tencent, Nippon Telegraph and Telephone Corporation (NTT), Yum China, Netease, vivo, Nutanix, and StreamNative. [Watch the sessions](https://www.youtube.com/playlist?list=PLqRma1oIkcWgeNb3kgzqFyg5sywH8S4yy).

To stay up to date with the latest events, you can join the following meetups or groups:

- [New York City Apache Pulsar Meetup](https://www.meetup.com/new-york-city-apache-pulsar-meetup/)
- [SF Bay Area Apache Pulsar Meetup](https://www.meetup.com/sf-bay-area-apache-pulsar-meetup/)
- [Netherlands Apache Pulsar Meetup](https://www.meetup.com/netherlands-apache-pulsar-meetup/)

> - [Japan Pulsar User Group](https://japan-pulsar-user-group.connpass.com/)

Reviewer: May we know why not add the following groups?

• [SoCal Apache Pulsar Neighborhood Meetup Group](https://www.meetup.com/socal-apache-pulsar-meetup-group/)
• [Beijing Apache Pulsar Meetup](https://www.meetup.com/beijing-apache-pulsar-meetup-by-streamnative/)
• [NorCal Apache Pulsar Neighborhood Meetup Group](https://www.meetup.com/nor-cal-apache-pulsar-meetup-group/)

Author: @tuhaihe

## Community growth

The development of the Pulsar community would not be possible without our contributors. Among them, Pulsar Committers and PMC members have taken the lead in contributing to and promoting the project. In 2022, many new faces joined the community while we also welcomed old friends taking more responsibility. Let’s take a look at the Pulsar community by numbers.

- **598** global contributors for the Pulsar main GitHub repository
- **12.1K+** GitHub stars
- **3.2K+** forks
- **16** new Committers

> - **5** new PMC members

Reviewer: 1. The NO. here is a bit confusing. Some are `total NO.` while others are `NO. of last year`. Does it make sense to add the `total NO. of committers and PMC members`? 

In this way, we can reduce users' confusion (data is consistent) and show the whole picture to them.

2. Does it make sense to add the `NO. of Pulsar releases` (show activeness and long history) and the `NO. of ecosystem plugins` (show diversity and compatibility)? 

These data can give users more trust and confidence in Pulsar.

Author: 1. This is a summary blog for 2022. The only reason I added a total number is that I am not sure the exact number of growth from last year so I added the total number. @tuhaihe If you have the number please let me know. Thanks.
2. For releases, people can find the info in the sections below. For ecosystem tools, you can have a look at the discussions in this pr. It is difficult to include all tools as we may have different vendors developing them.

- **9560** Pulsar Slack members
- **4.3K+** downloads
- **14M+** Docker pulls

The Pulsar community welcomes all kinds of contributions. For more information, see the [Apache Pulsar Contribution Guide](https://pulsar.apache.org/contribute/).

## Project releases

> In 2022, the Pulsar community worked hard to improve the project’s capabilities and fix existing bugs with 1 major version and 8 minor versions. Many important Pulsar capabilities were delivered in these releases, such as [cluster-level automatic failure recovery](https://github.com/apache/pulsar/pull/13316) and [lazy-loading added to PartitionedProducer](https://github.com/apache/pulsar/pull/10279). Check out each release below to see details.

Reviewer 2: ```suggestion
In 2022, the Pulsar community worked hard to improve the project’s capabilities and fix existing bugs with 1 major version and 8 minor versions. Many important Pulsar capabilities were delivered in these releases, such as [cluster-level automatic failure recovery](https://github.com/apache/pulsar/pull/13316) and [lazy-loading added to PartitionedProducer](https://github.com/apache/pulsar/pull/10279). Check out each release below to see details:
```

- [2.10.2](https://github.com/apache/pulsar/releases/tag/v2.10.2)
- [2.10.1](https://github.com/apache/pulsar/releases/tag/v2.10.1)
- [2.10.0](https://github.com/apache/pulsar/releases/tag/v2.10.0)
- [2.9.3](https://github.com/apache/pulsar/releases/tag/v2.9.3)
- [2.9.2](https://github.com/apache/pulsar/releases/tag/v2.9.2)
- [2.8.4](https://github.com/apache/pulsar/releases/tag/v2.8.4)
- [2.8.3](https://github.com/apache/pulsar/releases/tag/v2.8.3)
- [2.8.2](https://github.com/apache/pulsar/releases/tag/v2.8.2)
- [2.7.5](https://github.com/apache/pulsar/releases/tag/v2.7.5)

> For more information, see the [Release Notes page](https://pulsar.apache.org/release-notes/).

Reviewer 2: ```suggestion
For more information, see the [Release Notes page](pathname:///release-notes/).
```

... render an internal link for local debugging.

Updates about clients, Pulsar Manager, and Pulsar Helm Chart are listed below:

- [Pulsar C++ Client 3.1.0](https://github.com/apache/pulsar-client-cpp/releases/tag/v3.1.0)
- [Pulsar C++ Client 3.0.0](https://github.com/apache/pulsar-client-cpp/releases/tag/v3.0.0)
- [Pulsar Go Client 0.9.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.9.0)
- [Pulsar Go Client 0.8.1](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.1)
- [Pulsar Go Client 0.8.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.0)

> - [Pulsar Node.js client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)

Reviewer: ```suggestion
- [Pulsar Node.js Client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)
```

Keep consistent

> - [Pulsar Node.js client 1.6.2](https://github.com/apache/pulsar-client-node/releases/tag/v1.6.2)

Reviewer: ```suggestion
- [Pulsar Node.js Client 1.6.2](https://github.com/apache/pulsar-client-node/releases/tag/v1.6.2)
```

- [Pulsar Python Client 3.0.0](https://github.com/apache/pulsar-client-python/releases/tag/v3.0.0)

> - [Reactive client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.1.0)

Reviewer: ```suggestion
- [Reactive Client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.1.0)
```

- [Pulsar Manager 0.3.0](https://github.com/apache/pulsar-manager/releases/tag/v0.3.0)
- [Pulsar Helm Chart 3.0.0](https://github.com/apache/pulsar-helm-chart/releases/tag/pulsar-3.0.0)

> For more information, see the [Clients Release Notes page](https://pulsar.apache.org/release-notes/clients/).

Reviewer 2: ```suggestion
For more information, see the [Clients Release Notes page](pathname:///release-notes/clients/).
```

ditto

- [Spring for Apache Pulsar](https://spring.io/blog/2022/09/20/spring-for-apache-pulsar-0-1-0-m1-is-now-available): This project aims to provide Spring-friendly APIs, building blocks, and programming models for writing Java applications that interact with Apache Pulsar. For more information, read the blog [Spring for Apache Pulsar 0.1.0 available now](https://spring.io/blog/2022/12/15/spring-for-apache-pulsar-0-1-0-available-now).
- [Flink-Pulsar Sink Connector](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/datastream/pulsar/#pulsar-sink): The Flink-Pulsar Sink Connector is part of the Flink-Pulsar DataStream Connector. It implements Flink’s new [SinkV2 API](https://cwiki.apache.org/confluence/display/FLINK/FLIP-177%3A+Extend+Sink+API) and allows you to write Flink job results back to Pulsar topics seamlessly.
- [HTTP Sink Connector](https://github.com/apache/pulsar/issues/17719): The HTTP Sink Connector pushes records to any HTTP server with the record value in the body of POST method. The body of the HTTP request is the JSON representation of the record value.

> For more information, see the [Ecosystem page](https://pulsar.apache.org/ecosystem/).

Reviewer 2: ```suggestion
For more information, see the [Ecosystem page](pathname:///ecosystem/).
```

ditto

- **598** global contributors for the Pulsar main GitHub repository
- **12.1K+** GitHub stars
- **3.2K+** forks
- **16** new Committers
- **5** new PMC members
- **9560** Pulsar Slack members
- **4.3K+** downloads
- **14M+** Docker pulls

The Pulsar community welcomes all kinds of contributions. For more information, see the [Apache Pulsar Contribution Guide](https://pulsar.apache.org/contribute/).

## Project releases

In 2022, the Pulsar community worked hard to improve the project’s capabilities and fix existing bugs with 1 major version and 8 minor versions. Many important Pulsar capabilities were delivered in these releases, such as [cluster-level automatic failure recovery](https://github.com/apache/pulsar/pull/13316) and [lazy-loading added to PartitionedProducer](https://github.com/apache/pulsar/pull/10279). Check out each release below to see details.

> - [2.7.5](https://github.com/apache/pulsar/releases/tag/v2.7.5)

Reviewer 2: Use reverse order here? Then users can read the latest versions first.

Author: Changed.

- [2.8.2](https://github.com/apache/pulsar/releases/tag/v2.8.2)
- [2.8.3](https://github.com/apache/pulsar/releases/tag/v2.8.3)
- [2.8.4](https://github.com/apache/pulsar/releases/tag/v2.8.4)
- [2.9.2](https://github.com/apache/pulsar/releases/tag/v2.9.2)
- [2.9.3](https://github.com/apache/pulsar/releases/tag/v2.9.3)
- [2.10.0](https://github.com/apache/pulsar/releases/tag/v2.10.0)
- [2.10.1](https://github.com/apache/pulsar/releases/tag/v2.10.1)
- [2.10.2](https://github.com/apache/pulsar/releases/tag/v2.10.2)

For more information, see the [Release Notes page](https://pulsar.apache.org/release-notes/).

Updates about clients, Pulsar Manager, and Pulsar Helm Chart are listed below:

> - [Pulsar C++ Client 3.0.0](https://github.com/apache/pulsar-client-cpp/releases/tag/v3.0.0)

Reviewer 2: ditto

Author: Thanks. Changed.

- [Pulsar C++ Client 3.1.0](https://github.com/apache/pulsar-client-cpp/releases/tag/v3.1.0)
- [Pulsar Go Client 0.8.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.0)
- [Pulsar Go Client 0.8.1](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.1)
- [Pulsar Go Client 0.9.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.9.0)
- [Pulsar Node.js client 1.6.2](https://github.com/apache/pulsar-client-node/releases/tag/v1.6.2)
- [Pulsar Node.js client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)
- [Pulsar Python Client 3.0.0](https://github.com/apache/pulsar-client-python/releases/tag/v3.0.0)
- [Pulsar Manager 0.3.0](https://github.com/apache/pulsar-manager/releases/tag/v0.3.0)
- [Pulsar Helm Chart 3.0.0](https://github.com/apache/pulsar-helm-chart/releases/tag/pulsar-3.0.0)

> 

Reviewer 3: Please add a reference to the "Reactive client for Apache Pulsar 0.1.0" release:
```suggestion
 - [Reactive client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.1.0)
 
```

Author: Added. Thanks.

For more information, see the [Clients Release Notes page](https://pulsar.apache.org/release-notes/clients/).

## Pulsar ecosystem

> In 2022, the Pulsar community worked with other open-source communities to add more integrations to the Pulsar ecosystem. Notable integrations include:

Reviewer 4: I suggest to add the HTTP Sink connector introduced by [PIP 208](https://github.com/apache/pulsar/issues/17719)

Author: Added. Thanks.

> - [Spring for Apache Pulsar](https://spring.io/blog/2022/09/20/spring-for-apache-pulsar-0-1-0-m1-is-now-available): This project aims to provide Spring-friendly APIs, building blocks, and programming models for writing Java applications that interact with Apache Pulsar. Read this [blog](https://streamnative.io/blog/release/2022-09-21-announcing-spring-for-apache-pulsar/) to see details.

Reviewer 3: The StreamNative blog post is outdated. I'd suggest providing a link to the official Spring blog post about 0.1.0 release. /cc @cbornet 
```suggestion
- [Spring for Apache Pulsar](https://spring.io/blog/2022/09/20/spring-for-apache-pulsar-0-1-0-m1-is-now-available): This project aims to provide Spring-friendly APIs, building blocks, and programming models for writing Java applications that interact with Apache Pulsar. Please check the blog post [Spring for Apache Pulsar 0.1.0 available now](https://spring.io/blog/2022/12/15/spring-for-apache-pulsar-0-1-0-available-now) for most latest details and references to documentation.
```

Author: Updated. Thanks.

- [Delta Lake Sink Connector for Apache Pulsar](https://github.com/streamnative/pulsar-io-lakehouse): It is a Pulsar IO connector that pulls data from Apache Pulsar topics and persists data to Delta Lake. Read this [blog](https://streamnative.io/blog/release/2022-08-17-announcing-the-delta-lake-sink-connector-for-apache-pulsar/) to see details.
- [Iceberg Sink Connector for Apache Pulsar](https://github.com/streamnative/pulsar-io-lakehouse): It is a Pulsar IO connector that pulls data from Apache Pulsar topics and persists data to Iceberg tables. Read this [blog](https://streamnative.io/blog/release/2022-12-14-announcing-the-iceberg-sink-connector-for-apache-pulsar/) to see details.
- [Pulsar Resources Operator for Kubernetes](https://github.com/streamnative/pulsar-resources-operator): The Pulsar Resources Operator is an independent controller that automatically manages Pulsar resources on Kubernetes using manifest files. Read this [blog](https://streamnative.io/blog/release/2022-08-15-introducing-pulsar-resources-operator-for-kubernetes/) to see details.

## What’s next in 2023

> The Pulsar community seeks to further improve the project from different aspects. For example, we plan on releasing the first long-term support (LTS) version of Pulsar to meet the needs of different users. For those who want a more stable Pulsar version, they can choose the LTS version, which is expected to be released every 18 months with bug fix support for 24 months and security patches for 36 months. Ordinary releases will come out every 3 months with bug fix support and security patches for 6 months.

Reviewer 5: I wonder if it is worth changing the language here so that we do not over promise release frequency. The 2.11 release took 4 months, and it is only just wrapping up now. I know that we've discussed targeting frequent releases, and I know that users like predictability, but I think it is better to under promise release frequency here.

Author: Thanks. I removed the frequency info here.
